File: server/src/models/Payment.js
import mongoose from "mongoose";

const paymentSchema = new mongoose.Schema(
  {
    orderId: { type: mongoose.Schema.Types.ObjectId, ref: "Order", required: true, index: true },
    userId: { type: mongoose.Schema.Types.ObjectId, ref: "User", required: true, index: true },
    amount: { type: Number, required: true, min: 0 },
    provider: { type: String, default: "demo" },
    status: { type: String, enum: ["pending", "paid", "failed"], default: "pending" },
    transactionId: { type: String, required: true, unique: true, index: true },
  },
  { timestamps: true },
);

export const Payment = mongoose.model("Payment", paymentSchema);

File: microservices/payment-service/src/main/java/com/ecommerce/payment/model/Payment.java
package com.ecommerce.payment.model;

import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Document(collection = "payments")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Payment implements Serializable {

    @Id
    private String id;

    @Indexed
    private String orderId;

    @Indexed
    private String userId;

    private BigDecimal amount;

    @Builder.Default
    private String provider = "demo";

    @Builder.Default
    private String status = "pending";

    @Indexed(unique = true)
    private String transactionId;

    @CreatedDate
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;
}
